src/backend/question/views.py

```diff
@@ -40,8 +49,44 @@ def get_queryset(self):
             queryset = queryset.filter(category=category)
         if keypoint is not None:
             queryset = queryset.filter(keypoint=keypoint)
+        if solution_type is not None and solution_type != 'all':
+            queryset = queryset.filter(type=solution_type)
         return queryset
 
+    def get_queryset(self):
+        query_set = self.__get_query_set_helper()
+        limit = self.request.query_params.get('limit', 10)
```

Comment:
We can [Modifying the pagination style](https://www.django-rest-framework.org/api-guide/pagination/#modifying-the-pagination-style) here